### PR TITLE
Support unified cluster-aws app

### PR DESCRIPTION
**Pull request by nprokopic**
### What does this PR do?

Towards https://github.com/giantswarm/roadmap/issues/3119

With cluster-aws v0.76.0 and newer, default apps are deployed with cluster-aws and default-apps-aws app is not deployed anymore.

Note: The latest cluster-aws release is v0.75.0 and the [default apps addition](https://github.com/giantswarm/cluster-aws/pull/581) is not yet merged (but it has been tested and it's ready to be merged). This PR assumes that the [default apps addition](https://github.com/giantswarm/cluster-aws/pull/581) will be merged and that v0.76.0 will be released.

⚠️ This should be used in cluster-test-suites only after cluster-aws v0.76.0 has been released. Otherwise cluster-test-suites will fail.

### Checklist

- [x] CHANGELOG.md has been updated

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.21
 require (
 	dario.cat/mergo v1.0.0
 	github.com/giantswarm/apiextensions-application v0.6.1
-	github.com/giantswarm/clustertest v0.20.0
+	github.com/giantswarm/clustertest v0.20.1-0.20240512235412-f3c924c97b9b
```
Comment (nprokopic) [inline go.mod:8]: TODO: update after merging and releasing https://github.com/giantswarm/clustertest/pull/181

File: pkg/standup/standup.go [hunk 2]
```diff
@@ -58,6 +59,12 @@ func (c *Client) Standup(cluster *application.Cluster) (*application.Cluster, er
 	applyCtx, cancelApplyCtx := context.WithTimeout(ctx, 20*time.Minute)
 	defer cancelApplyCtx()
 
+	skipsDefaultApps, err := cluster.UsesUnifiedClusterApp()
+	Expect(err).NotTo(HaveOccurred())
+	if skipsDefaultApps {
+		logger.Log("Deploying only unified %s app (with default apps) and skipping %s app.", cluster.ClusterApp.AppName, cluster.DefaultAppsApp.AppName)
+	}
+
```
Comment (AverageMarcus) [inline pkg/standup/standup.go:67]: I think we can leave this out to be honest. It's just extra code we'll need to remove when all providers are complete.

File: pkg/clusterbuilder/providers/capa/values/cluster_values.yaml [hunk 3]
```diff
@@ -23,3 +23,8 @@ global:
       spotInstances:
         enabled: true
         maxPrice: 0.2960
+
+  apps:
+    externalDns:
+      values:
+        triggerLoopOnEvent: true
```
Comment (AndiDog) [inline pkg/clusterbuilder/providers/capa/values/cluster_values.yaml:30]: Is this related to the unification? Without context, this feels like a general improvement to reconcile faster?!

File: pkg/clusterbuilder/providers/capa/values/private-cluster_values.yaml [hunk 4]
```diff
@@ -57,3 +57,18 @@ global:
...
+            eks.amazonaws.com/role-arn: "{{ .ClusterName }}-CertManager-Role"
+        giantSwarmClusterIssuer:
+          acme:
+            http01:
+              enabled: false
+            dns01:
+              route53:
+                enabled: true
+                # TODO Use a variable as soon as it is available.
+                region: "eu-north-1"
```
Comment (AndiDog) [inline pkg/clusterbuilder/providers/capa/values/private-cluster_values.yaml:74]: Does this do anything? Route53 is global.

The TODO comment isn't actionable for people without context.

File: pkg/clusterbuilder/providers/capa/values/cluster_values.yaml (hunk 3 above)
Comment (AverageMarcus) [reply pkg/clusterbuilder/providers/capa/values/cluster_values.yaml:30]: This and the other bit you commented on are copied over from the existing values for default-apps (as they now need to be defined on the cluster app values instead)

Comment (nprokopic) [reply pkg/clusterbuilder/providers/capa/values/cluster_values.yaml:30]: What Marcus said ☝️

File: pkg/standup/standup.go (hunk 2 above)
Comment (nprokopic) [reply pkg/standup/standup.go:67]: Sure, will remove it.

File: pkg/clusterbuilder/providers/capa/values/private-cluster_values.yaml (hunk 4 above)
Comment (nprokopic) [reply pkg/clusterbuilder/providers/capa/values/private-cluster_values.yaml:74]: I just moved the config from default apps config to here https://github.com/giantswarm/cluster-standup-teardown/pull/20#discussion_r1598135566

Comment (nprokopic) [reply pkg/clusterbuilder/providers/capa/values/private-cluster_values.yaml:74]: No idea why this is here TBH

File: pkg/standup/standup.go (hunk 2 above)
Comment (nprokopic) [reply pkg/standup/standup.go]: Removed.

File: go.mod (hunk 1 above)
Comment (nprokopic) [reply go.mod]: Updated.

**Review by AverageMarcus — APPROVED** [review]
